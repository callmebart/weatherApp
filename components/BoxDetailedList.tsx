import { View, Text, StyleSheet } from 'react-native'
import globalStore from "../store/GlobalSettings";
import TextTicker from 'react-native-text-ticker';
import { detailedContent } from "../consts/en";
import getRecalculatedValue from "../utils/getRecalculatedValue";
import { useTheme } from "../hooks/ThemeProvider";
import { themeMode } from "../utils/themeMode";

export default function BoxDetailedList({ item }: any) {
    const unit = getRecalculatedValue(item.main.feels_like);
    const { theme } = useTheme();
    const styles = useStyles(themeMode[theme].color);

    return (

        <View style={styles.dataListStyles}>
            <View style={styles.row}>
                <View style={styles.inRowBox}>
                    <TextTicker
                        style={[styles.valueTitle, themeMode[theme]]}
                        duration={4000}
                        loop
                        scroll
                        repeatSpacer={100}
                        marqueeDelay={1200}
                    >
                        {detailedContent.per_temp}
                    </TextTicker>
                    <Text style={[styles.textAligement, styles.value]}>
                        {unit} {globalStore.unit}
                    </Text>
                </View>
                <View style={styles.inRowBox}>
                    <Text style={[styles.textAligement, styles.valueTitle]}>Humidity</Text>
                    <Text style={[styles.textAligement, styles.value]}>{item.main.humidity} %</Text>
                </View>
            </View>

            <View style={styles.row}>
                <View style={styles.inRowBox}>
                    <Text style={[styles.textAligement, styles.valueTitle]}>{detailedContent.wind}</Text>
                    <Text style={[styles.textAligement, styles.value]}>{item.wind.speed} km/h</Text>
                </View>
                <View style={styles.inRowBox}>
                    <TextTicker
                        style={[styles.textAligement, styles.valueTitle]}
                        duration={4000}
                        loop
                        bounce
                        scroll
                        repeatSpacer={100}
                        marqueeDelay={1200}
                    >
                        {detailedContent.probability}
                    </TextTicker>
                    <Text style={[styles.textAligement, styles.value]}>{item.pop === 0 ? 0 : 100} %</Text>
                </View>
            </View>

            <View style={styles.row}>
                <View style={styles.inRowBox}>
                    <Text style={[styles.textAligement, styles.valueTitle]}>{detailedContent.pressure}</Text>
                    <Text style={[styles.textAligement, styles.value]}>{item.main.pressure} hPa</Text>
                </View>
                <View style={styles.inRowBox}>
                    <Text style={[styles.textAligement, styles.valueTitle]}>{detailedContent.cloudiness}</Text>
                    <Text style={[styles.textAligement, styles.value]}>{item.clouds.all} %</Text>
                </View>
            </View>
        </View>
    )
}

const useStyles = (color: string) => StyleSheet.create({
    row: {
        flexDirection: "row",
        justifyContent: "space-between",
    },
    textAligement: {
        textAlign: "center",
        color: color,
    },
    value: {
        fontSize: 25,
        fontWeight: "bold",
    },
    valueTitle: {
        fontSize: 18,
    },
    inRowBox: {
        width: 160,
        justifyContent: "center",
        alignItems: "center",
        margin: 10,
    },
    dataListStyles: {
        flex: 1,
    },
})