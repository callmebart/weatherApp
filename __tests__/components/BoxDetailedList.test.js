import { render } from "@testing-library/react-native";
import BoxDetailedList from "../../components/BoxDetailedList";

describe('BoxDetail', () => {
  it('should render', () => {
    const item ={"clouds": {"all": 89}, "dt": 1681797600, "dt_txt": "2023-04-18 06:00:00", "groupKey": "2023-04-18", "main": {"feels_like": 4.5, "grnd_level": 991, "humidity": 85, "pressure": 1020, "sea_level": 1020, "temp": 7.76, "temp_kf": 0, "temp_max": 7.76, "temp_min": 7.76}, "pop": 0, "sys": {"pod": "d"}, "visibility": 10000, "weather": [{"description": "overcast clouds", "icon": "04d", "id": 804, "main": "Clouds"}], "wind": {"deg": 70, "gust": 10.72, "speed": 5.73}}
    const root  = render(<BoxDetailedList item={item}/>).toJSON();
    expect(root).toMatchSnapshot();
  });
});