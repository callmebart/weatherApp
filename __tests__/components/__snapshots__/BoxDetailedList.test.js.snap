// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxDetail should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "flex": undefined,
            },
          ]
        }
      >
        <Text
          animationType="auto"
          bounce={true}
          bounceDelay={0}
          bounceSpeed={50}
          duration={4000}
          easing={[Function]}
          isInteraction={true}
          loop={true}
          marqueeDelay={1200}
          marqueeOnMount={true}
          numberOfLines={1}
          scrollSpeed={150}
          style={
            Array [
              Array [
                Object {
                  "fontSize": 18,
                },
                Object {
                  "color": "white",
                },
              ],
              Object {
                "opacity": 0,
              },
            ]
          }
          useNativeDriver={true}
        >
          Perceived temperature
        </Text>
        <RCTScrollView
          display="flex"
          horizontal={true}
          onContentSizeChange={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          scrollEnabled={false}
          scrollEventThrottle={16}
          showsHorizontalScrollIndicator={false}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              false,
            ]
          }
        >
          <View>
            <Text
              animationType="auto"
              bounce={true}
              bounceDelay={0}
              bounceSpeed={50}
              collapsable={false}
              duration={4000}
              easing={[Function]}
              forwardedRef={[Function]}
              isInteraction={true}
              loop={true}
              marqueeDelay={1200}
              marqueeOnMount={true}
              numberOfLines={1}
              scrollSpeed={150}
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": null,
                }
              }
              useNativeDriver={true}
            >
              Perceived temperature
            </Text>
          </View>
        </RCTScrollView>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        4
         
        Â°C
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Humidity
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        85
         %
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Wind N
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        5.73
         km/h
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "flex": undefined,
            },
          ]
        }
      >
        <Text
          animationType="auto"
          bounce={true}
          bounceDelay={0}
          bounceSpeed={50}
          duration={4000}
          easing={[Function]}
          isInteraction={true}
          loop={true}
          marqueeDelay={1200}
          marqueeOnMount={true}
          numberOfLines={1}
          scrollSpeed={150}
          style={
            Array [
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 18,
                },
              ],
              Object {
                "opacity": 0,
              },
            ]
          }
          useNativeDriver={true}
        >
          Probability of precipitation
        </Text>
        <RCTScrollView
          display="flex"
          horizontal={true}
          onContentSizeChange={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          scrollEnabled={false}
          scrollEventThrottle={16}
          showsHorizontalScrollIndicator={false}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              false,
            ]
          }
        >
          <View>
            <Text
              animationType="auto"
              bounce={true}
              bounceDelay={0}
              bounceSpeed={50}
              collapsable={false}
              duration={4000}
              easing={[Function]}
              forwardedRef={[Function]}
              isInteraction={true}
              loop={true}
              marqueeDelay={1200}
              marqueeOnMount={true}
              numberOfLines={1}
              scrollSpeed={150}
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "textAlign": "center",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": null,
                }
              }
              useNativeDriver={true}
            >
              Probability of precipitation
            </Text>
          </View>
        </RCTScrollView>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        0
         %
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Pressure
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        1020
         hPa
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Cloudiness
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
            },
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        89
         %
      </Text>
    </View>
  </View>
</View>
`;
